core/modules/user/user.module

```diff
@@ -853,7 +853,7 @@ function user_validate_current_pass(&$form, &$form_state) {
     // that prevent them from being empty if they are changed.
     if ((strlen(trim($form_state['values'][$key])) > 0) && ($form_state['values'][$key] != $account->$key)) {
       require_once BACKDROP_ROOT . '/' . settings_get('password_inc', 'core/includes/password.inc');
-      $current_pass_failed = empty($form_state['values']['current_pass']) || !user_check_password($form_state['values']['current_pass'], $account);
+      $current_pass_failed = strlen(trim($form_state['values']['current_pass'])) == 0 || !user_check_password($form_state['values']['current_pass'], $account);
```

Comment:
In the respective commit in d.org, there was another single-line change in the `user_save()` function that you seem to be missing @jenlampton

Reply:
I don't think "empty" is a good a check as "length = 0", this is still an improvement.